### PR TITLE
Token deposit locker

### LINKED ISSUE #619: Deposit locker can lock TLN
Deposit locker can receive TLN from the auction contract and lock them.
Users can withdraw TLN from deposit locker after lock period.

### PR DESCRIPTION
**Pull request by berndbohmeier**
closes #619

### CODE REVIEW

File: contracts/contracts/auction/TokenDepositLocker.sol [hunk 1]
```diff
@@ -0,0 +1,36 @@
+pragma solidity ^0.5.8;
+
+import "./DepositLocker.sol";
+import "../token/TrustlinesNetworkToken.sol";
+
+/*
+  The ETHDepositLocker contract locks ETH deposits
```
Comment (cducrest) [inline contracts/contracts/auction/TokenDepositLocker.sol:7]: The TokenDepositLocker contract locks Token deposits

File: contracts/contracts/auction/TokenDepositLocker.sol [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  For more information see DepositLocker.sol
+*/
+
+contract TokenDepositLocker is DepositLocker {
+    TrustlinesNetworkToken public token;
```
Comment (cducrest) [inline contracts/contracts/auction/TokenDepositLocker.sol:13]: It should be better to use a token interface to keep it generic as it was the mentality we had on these contracts

File: contracts/contracts/auction/TokenDepositLocker.sol [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    constructor(TrustlinesNetworkToken _trustlinesNetworkToken) public {
```
Comment (cducrest) [inline contracts/contracts/auction/TokenDepositLocker.sol:15]: I think it is weird that the contract now has both a `constructor` and an `init`, I feel like it is less obvious to work with

File: contracts/tests/test_deposit_locker.py [hunk 4]
```diff
@@ -220,11 +138,43 @@ def withdraw(self, depositor):
...
+            token_symbol,
+            token_decimal,
+            premint_token_address,
+            premint_token_value,
+        )
+
+        token_contract = deploy_contract(
+            "TrustlinesNetworkToken", constructor_args=constructor_args
+        )
+        constructor_args = (token_contract.address,)
```
Comment (cducrest) [inline contracts/tests/test_deposit_locker.py:175]: I would make the difference between deposit_constructor_args and token_constructor_args

File: contracts/contracts/auction/TokenDepositLocker.sol (hunk 1 above)
Comment (berndbohmeier) [reply contracts/contracts/auction/TokenDepositLocker.sol:7]: Changed

File: contracts/contracts/auction/TokenDepositLocker.sol (hunk 3 above)
Comment (berndbohmeier) [reply contracts/contracts/auction/TokenDepositLocker.sol]: Changed to only use init.

File: contracts/tests/test_deposit_locker.py (hunk 4 above)
Comment (berndbohmeier) [reply contracts/tests/test_deposit_locker.py]: outdated now, as there is not deposit constructor anymore.

**Pull request merged**